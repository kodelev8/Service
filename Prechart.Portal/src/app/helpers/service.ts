import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AccountService } from "../shared/account/services/account.service";
import { UrlHelper } from "./url";
import {User} from "../shared/account/models/user";

@Injectable({ providedIn: 'root' })
export class ServiceHelper {
  headers: HttpHeaders;
  constructor(
    private http: HttpClient,
    private accountService: AccountService,
    private urlHelper: UrlHelper)
  {
    if (this.accountService.userValue) {
      var token = this.accountService.userValue;

      this.headers = new HttpHeaders({
        'Authorization': `Bearer ${token}`,
      });
    }
  }

  get(port: string, path: string) {
    var url = this.urlHelper.format(port, path);
    return this.http.get<any>(url, { headers: this.headers });
  }

  post(port: string, path: string, object: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, { headers: this.headers });
  }

  validateUser(port: string, path: string, object: any, options: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, options);
  }
}
