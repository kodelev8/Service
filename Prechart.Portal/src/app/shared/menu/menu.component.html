<p-menubar [model]="menuItems">
  <ng-template pTemplate="start">
    <a class="psp-brand " href="#"><img src="assets/images/logo/logo.svg" alt="Prechart Software" class="psp-logo" /></a>
  </ng-template>
  <ng-template pTemplate="end">
    <p-button  label="Loguit" icon="pi pi-power-off" (onClick)="logout()"></p-button>
  </ng-template>
</p-menubar>
