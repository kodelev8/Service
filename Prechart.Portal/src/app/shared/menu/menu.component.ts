import { Component, OnInit } from '@angular/core';
import { MenuItem } from 'primeng/api';
import { AccountService } from '../account/services/account.service';

@Component({
  selector: 'psp-menu',
  templateUrl: './menu.component.html',
})
export class MenuComponent implements OnInit {
  title = 'Menu';
  user: string;

  constructor(private accountService: AccountService) {
    this.accountService.user.subscribe(x => this.user = x);
  }

  menuItems: MenuItem[];

  ngOnInit() {
    this.menuItems = [
      //{
      //  label: 'DTR',
      //  routerLink: '/timerecord'
      //},
      {
        label: 'Belastingen',
        items: [{
          label: 'Belasting Tabellen Wit/Groen',
          items: [
            {
              label: 'Uploaden',
              routerLink: '/belastingen/upload'
            },
            {
              label: 'Zoek',
              routerLink: '/belastingen/search'
            }
          ]
        }]
      },
      {
        label: 'Employee',
        items: [
          {          
            label: 'Uploaden',
            routerLink: '/employee/upload'
          },
          {
            label: 'Zoek',
            routerLink: '/employee/search'
          },
          {
            label: 'Daywage',
            routerLink: '/employee/daywage'
          }
        ],
      }, {
        label: 'Werkgever',
        items: [
          {
            label: 'Beheer',
            routerLink: '/werkgever/management'
          },
          {
            label: 'Collectieve Aangifte ',
            routerLink: '/werkgever/collective'
          },
        ],
      },
    ];
  }

  logout() {
    this.accountService.logout();
  }
}
