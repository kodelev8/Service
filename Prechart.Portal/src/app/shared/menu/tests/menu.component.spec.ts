import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { MenubarModule } from 'primeng/menubar';

import { MenuComponent } from 'src/app/shared/menu/menu.component';
import { AccountService } from 'src/app/shared/account/services/account.service'
import { HttpClientModule } from '@angular/common/http';

describe('MenuComponent', () => {
  let fixture: ComponentFixture<MenuComponent>;
  let component: MenuComponent;
  let element: any;
  let constructoMock: any;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MenubarModule,
        HttpClientModule,
      ],
      declarations: [
        MenuComponent,
      ],
      providers: [AccountService],
    }).compileComponents();    
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(MenuComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    fixture.detectChanges();
  });

  it('should create the Menu page', () => {    
    expect(component).toBeTruthy();
  });

  it('should render the Menu page correctly', () => {    
    expect(element).toMatchSnapshot();
  });

  it('should render the menu items correctly', () => {
    let element = fixture.debugElement.query(By.css('.p-menuitem-text'));
    let span = element.nativeElement;
    
    expect(span.innerHTML).toContain('Belastingen');
    expect(element).toMatchSnapshot();
  });

});
