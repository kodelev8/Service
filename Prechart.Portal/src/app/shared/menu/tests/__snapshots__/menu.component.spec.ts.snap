// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent should render the Menu page correctly 1`] = `
<div
  id="root1"
>
  <p-menubar
    class="p-element"
  >
    <div
      class="p-menubar p-component"
    >
      
      <a
        class="p-menubar-button"
        tabindex="0"
      >
        <i
          class="pi pi-bars"
        />
      </a>
      <p-menubarsub
        class="p-element"
        root="root"
      >
        <ul
          class="p-menubar-root-list"
          role="menubar"
        >
          
          <li
            class="p-element p-menuitem"
            ptooltip=""
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="false"
              class="p-ripple p-element p-menuitem-link"
              pripple=""
              role="menuitem"
              tabindex="0"
              target="undefined"
            >
              
              <span
                class="p-menuitem-text"
              >
                Belastingen
              </span>
              
              
              <span
                class="p-submenu-icon pi pi-angle-down"
              />
              
            </a>
            
            
            <p-menubarsub
              class="p-element"
            >
              <ul
                class="p-submenu-list"
                role="menu"
              >
                
                <li
                  class="p-element p-menuitem"
                  ptooltip=""
                  role="none"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="p-ripple p-element p-menuitem-link"
                    pripple=""
                    role="menuitem"
                    tabindex="0"
                    target="undefined"
                  >
                    
                    <span
                      class="p-menuitem-text"
                    >
                      Belasting Tabellen Wit/Groen
                    </span>
                    
                    
                    <span
                      class="p-submenu-icon pi pi-angle-right"
                    />
                    
                  </a>
                  
                  
                  <p-menubarsub
                    class="p-element"
                  >
                    <ul
                      class="p-submenu-list"
                      role="menu"
                    >
                      
                      <li
                        class="p-element p-menuitem"
                        ptooltip=""
                        role="none"
                      >
                        
                        <a
                          class="p-ripple p-element p-menuitem-link"
                          href="/belastingen/upload"
                          pripple=""
                          role="menuitem"
                          tabindex="0"
                        >
                          
                          <span
                            class="p-menuitem-text"
                          >
                            Uploaden
                          </span>
                          
                          
                          
                        </a>
                        
                        
                      </li>
                      
                      
                      <li
                        class="p-element p-menuitem"
                        ptooltip=""
                        role="none"
                      >
                        
                        <a
                          class="p-ripple p-element p-menuitem-link"
                          href="/belastingen/search"
                          pripple=""
                          role="menuitem"
                          tabindex="0"
                        >
                          
                          <span
                            class="p-menuitem-text"
                          >
                            Zoek
                          </span>
                          
                          
                          
                        </a>
                        
                        
                      </li>
                      
                      
                    </ul>
                  </p-menubarsub>
                  
                </li>
                
                
              </ul>
            </p-menubarsub>
            
          </li>
          
          
          <li
            class="p-element p-menuitem"
            ptooltip=""
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="false"
              class="p-ripple p-element p-menuitem-link"
              pripple=""
              role="menuitem"
              tabindex="0"
              target="undefined"
            >
              
              <span
                class="p-menuitem-text"
              >
                Employee
              </span>
              
              
              <span
                class="p-submenu-icon pi pi-angle-down"
              />
              
            </a>
            
            
            <p-menubarsub
              class="p-element"
            >
              <ul
                class="p-submenu-list"
                role="menu"
              >
                
                <li
                  class="p-element p-menuitem"
                  ptooltip=""
                  role="none"
                >
                  
                  <a
                    class="p-ripple p-element p-menuitem-link"
                    href="/employee/upload"
                    pripple=""
                    role="menuitem"
                    tabindex="0"
                  >
                    
                    <span
                      class="p-menuitem-text"
                    >
                      Uploaden
                    </span>
                    
                    
                    
                  </a>
                  
                  
                </li>
                
                
                <li
                  class="p-element p-menuitem"
                  ptooltip=""
                  role="none"
                >
                  
                  <a
                    class="p-ripple p-element p-menuitem-link"
                    href="/employee/search"
                    pripple=""
                    role="menuitem"
                    tabindex="0"
                  >
                    
                    <span
                      class="p-menuitem-text"
                    >
                      Zoek
                    </span>
                    
                    
                    
                  </a>
                  
                  
                </li>
                
                
              </ul>
            </p-menubarsub>
            
          </li>
          
          
          <li
            class="p-element p-menuitem"
            ptooltip=""
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="false"
              class="p-ripple p-element p-menuitem-link"
              pripple=""
              role="menuitem"
              tabindex="0"
              target="undefined"
            >
              
              <span
                class="p-menuitem-text"
              >
                Werkgever
              </span>
              
              
              <span
                class="p-submenu-icon pi pi-angle-down"
              />
              
            </a>
            
            
            <p-menubarsub
              class="p-element"
            >
              <ul
                class="p-submenu-list"
                role="menu"
              >
                
                <li
                  class="p-element p-menuitem"
                  ptooltip=""
                  role="none"
                >
                  
                  <a
                    class="p-ripple p-element p-menuitem-link"
                    href="/werkgever/management"
                    pripple=""
                    role="menuitem"
                    tabindex="0"
                  >
                    
                    <span
                      class="p-menuitem-text"
                    >
                      Beheer
                    </span>
                    
                    
                    
                  </a>
                  
                  
                </li>
                
                
                <li
                  class="p-element p-menuitem"
                  ptooltip=""
                  role="none"
                >
                  
                  <a
                    class="p-ripple p-element p-menuitem-link"
                    href="/werkgever/collective"
                    pripple=""
                    role="menuitem"
                    tabindex="0"
                  >
                    
                    <span
                      class="p-menuitem-text"
                    >
                      Collectieve Aangifte 
                    </span>
                    
                    
                    
                  </a>
                  
                  
                </li>
                
                
              </ul>
            </p-menubarsub>
            
          </li>
          
          
        </ul>
      </p-menubarsub>
      <div
        class="p-menubar-end"
      >
        
        
      </div>
      
      
    </div>
  </p-menubar>
</div>
`;

exports[`MenuComponent should render the menu items correctly 1`] = `
DebugElement {
  "nativeNode": <span
    class="p-menuitem-text"
  >
    Belastingen
  </span>,
}
`;
